request-manager/request/manager.go

```diff
@@ -353,6 +394,29 @@ func (m *manager) Status(requestId string) (proto.RequestStatus, error) {
 	return reqStatus, nil
 }
 
+func (m *manager) Finish(requestId string, finishParams proto.FinishRequestParams) error {
+	req, err := m.Get(requestId)
+	if err != nil {
+		return err
+	}
+
+	now := time.Now()
```

Comment:
Should JR set finish time? If there's a delay sending finished req back to RM, this will skew the finish time.

Reply:
Yeah, I guess it makes sense to have the finish time be as close as possible to the end of the last job, especially now that the JR retries the FinishRequest call several times.